cmd/ocm-backplane/remediation/remediation.go

```diff
@@ -0,0 +1,343 @@
+	rc.CurrentContext = targetContextNickName
+
+	logger.Debugln("Saving new API config")
+	// Save the config
+	if err = login.SaveKubeConfig(clusterID, rc, false, ""); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func runDeleteRemediation(args []string, clusterKey string, urlFlag string) error {
+	// ======== Initialize backplaneURL ========
+	bpConfig, err := config.GetBackplaneConfiguration()
+	if err != nil {
+		return err
+	}
+
+	bpCluster, err := utils.DefaultClusterUtils.GetBackplaneCluster(clusterKey)
+	if err != nil {
+		return err
+	}
+
+	backplaneHost := bpConfig.URL
+
+	clusterID := bpCluster.ClusterID
+
+	if urlFlag != "" {
+		backplaneHost = urlFlag
+	}
+
+	// ======== Parsing Args ========
+	if len(args) < 1 {
+		return fmt.Errorf("missing remediations service account name as an argument")
+	}
+	remediationSA := args[0]
+
+	accessToken, err := ocm.DefaultOCMInterface.GetOCMAccessToken()
+	if err != nil {
+		return err
+	}
+	// ======== Call Endpoint ========
+	err = doDeleteRemediation(backplaneHost, clusterID, *accessToken, remediationSA)
+	// ======== Render Results ========
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Deleted remediation RBAC on cluster %s\n", clusterID)
+	return nil
+}
+
+// TODO are we missusing the backplane api package? We have generated functions like backplaneapi.CreateRemediationWithResponse which already reads the body. All the utils functions do read the body again. Failing my calls here.
+func doCreateRemediation(api string, clusterID string, accessToken string, remediationName string) (proxyURI string, err error) {
+	client, err := backplaneapi.DefaultClientUtils.MakeBackplaneAPIClientWithAccessToken(api, accessToken)
+	if err != nil {
+		return "", fmt.Errorf("unable to create backplane api client")
+	}
+
+	logger.Debug("Sending request...")
+	resp, err := client.CreateRemediationWithResponse(context.TODO(), clusterID, &BackplaneApi.CreateRemediationParams{Remediation: remediationName})
+	if err != nil {
+		logger.Debug("unexpected...")
+		return "", err
+	}
+
+	// TODO figure out the error handling here
+	if resp.StatusCode() != http.StatusOK {
+		// logger.Debugf("Unmarshal error resp body: %s", resp.Body)
+		var dest BackplaneApi.Error
+		if err := json.Unmarshal(resp.Body, &dest); err != nil {
+			// Avoid squashing the HTTP response info with Unmarshal err...
+			logger.Debugf("Unmarshaled %s", *dest.Message)
+
+			bodyStr := strings.ReplaceAll(string(resp.Body[:]), "\n", " ")
+			err := fmt.Errorf("code:'%d'; failed to unmarshal response:'%s'; %w", resp.StatusCode(), bodyStr, err)
+			return "", err
+		}
+		return "", errors.New(*dest.Message)
+
+	}
+	return api + *resp.JSON200.ProxyUri, nil
+}
+
+// CreateRemediationWithConn can be used to programtically interact with backplaneapi
+func CreateRemediationWithConn(bp config.BackplaneConfiguration, ocmConnection *ocmsdk.Connection, clusterID string, remediationName string) (config *rest.Config, serviceAccountName string, err error) {
```

Comment:
Agreed! Moved to pkg/remediation
I did not move the unit test as they are mocking the command mostly and should reasonably cover the remediation pkg 

The Create|DeleteRemediationWithConn  functions are not tested currently but I think we do not need them actually.
While trying to test/mock/deal with them, I realize this is code leftover from an earlier draft for the same remediation feature. 
There is really no difference other then give the access token per parameter or give the ocm connection and then building a rest config instead of return the proxyUrl 
I think  this will need careful examination of what to export and what to use in CAD/osdctl which both use this as a dependecy ( which is already strange enough, there should be some kind of library and backplane-cli should pull from there as well ) https://github.com/openshift/osdctl/blob/f475456b95544fce30378f4a7684903e0626efb1/cmd/common/helpers.go#L7 

getting late, excuse the confusing comment, this is mostly to capture my thought for future 
TLDR will take another look :D